operator/pkg/manifest/operations.go

```diff
@@ -110,44 +116,47 @@ func NewOperations(logger *logr.Logger, restConfig *rest.Config, releaseName str
 	return operations, nil
 }
 
-func (o *Operations) getClusterResources(deployInfo InstallInfo, operation HelmOperation) (kube.ResourceList,
-	kube.ResourceList, error,
-) {
+func (o *Operations) getClusterResources(deployInfo InstallInfo, operation HelmOperation) (ResourceLists, error) {
 	if deployInfo.ChartPath == "" {
 		if err := o.repoHandler.Add(deployInfo.RepoName, deployInfo.URL, o.logger); err != nil {
-			return nil, nil, err
+			return ResourceLists{}, err
 		}
 	}
 
 	manifest, err := o.getManifestForChartPath(deployInfo.ChartPath, deployInfo.ChartName, o.actionClient, o.args)
 	if err != nil {
-		return nil, nil, err
+		return ResourceLists{}, err
 	}
 
-	if err = o.helmClient.HandleNamespace(o.actionClient, operation); err != nil {
-		return nil, nil, err
+	resourceListWithNs, err := o.helmClient.HandleNamespace(o.actionClient, operation)
+	if err != nil {
+		return ResourceLists{}, err
 	}
 
 	targetResources, err := o.helmClient.GetTargetResources(deployInfo.Ctx, manifest,
 		o.actionClient.Namespace, o.resourceTransforms, deployInfo.BaseResource)
 	if err != nil {
-		return nil, nil, err
+		return ResourceLists{}, fmt.Errorf("could not render resources from manifest: %w", err)
 	}
 
 	existingResources, err := util.FilterExistingResources(targetResources)
 	if err != nil {
-		return nil, nil, errors.Wrap(err, "could not render current resources from manifest")
+		return ResourceLists{}, fmt.Errorf("could not render existing resources from manifest: %w", err)
 	}
 
-	return targetResources, existingResources, nil
+	return ResourceLists{
+		Target:    targetResources,
+		Installed: existingResources,
+		Namespace: resourceListWithNs,
+	}, nil
 }
 
 func (o *Operations) VerifyResources(deployInfo InstallInfo) (bool, error) {
-	targetResources, existingResources, err := o.getClusterResources(deployInfo, "")
+	resourceLists, err := o.getClusterResources(deployInfo, "")
 	if err != nil {
 		return false, errors.Wrap(err, "could not render current resources from manifest")
 	}
-	if len(targetResources) > len(existingResources) {
+	if len(resourceLists.Target) > len(resourceLists.Installed) {
```

Comment:
Can we add a comment here to explain why the amount of Target > installed consider invalid? what happens if Target < Installed?

Reply:
Target can never be less than installed resources. I could add a comment but it‘s by the nature of installed list of resources. If resources are missing they‘re added back.